.github/workflows/ci.yml

```diff
@@ -98,63 +98,49 @@ jobs:
       - run: ./scalafmt --test
       - run: yarn install
       - run: yarn format-check
-  native-image:
+  scala-native:
     permissions:
       contents: write  # for actions/upload-release-asset to upload release asset
-    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [macOS-latest, ubuntu-latest]
-        libc: [default, musl]
-        exclude:
-          - os: macOS-latest
-            libc: musl
-        include:
-          - os: macOS-latest
-            artifact: scalafmt-macos
-            env:
-              NATIVE_IMAGE_STATIC: none
-          - os: ubuntu-latest
-            libc: default
-            artifact: scalafmt-linux-glibc
-            env:
-              NATIVE_IMAGE_STATIC: nolibc
-          - os: ubuntu-latest
-            libc: musl
-            artifact: scalafmt-linux-musl
-            env:
-              NATIVE_IMAGE_STATIC: musl
-    env: ${{ matrix.env }}
+        deploy: [ 
+          { os : macOS-13, name: scalafmt-x86_64-apple-darwin, path: "scalafmt-cli/native/target/scala-2.13/scalafmt-cli"}, 
```

Comment:
i just thought that perhaps all these OS and architecture values can be easily deduced from the environment and don't have to be hardcoded...

also, should we just call sbt and ask for the location of the target directory, instead of hardcoding with `scala-2.13`?

if that can be done, then i am ok without a script.